var/spack/repos/builtin/packages/nccmp/package.py

```diff
@@ -0,0 +1,30 @@
+from spack import *
+import os
+
+class Nccmp(Package):
+    """Compare NetCDF Files"""
+    homepage = "http://nccmp.sourceforge.net/"
+    url      = "http://downloads.sourceforge.net/project/nccmp/nccmp-1.8.2.0.tar.gz"
+
+    version('1.8.2.0', '81e6286d4413825aec4327e61a28a580')
+
+    depends_on('netcdf')
+
+    def install(self, spec, prefix):
+        # Configure says: F90 and F90FLAGS are replaced by FC and
+        # FCFLAGS respectively in this configure, please unset
+        # F90/F90FLAGS and set FC/FCFLAGS instead and rerun configure
+        # again.
+        os.environ['FC'] = os.environ['F90']
+        del os.environ['F90']
+        try:
+            os.environ['FCFLAGS'] = os.environ['F90FLAGS']
+            del os.environ['F90FLAGS']
+        except KeyError:    # There are no flags
```

Comment:
OK, I will change as you suggest.

In the process, I will push back against (non-standard) Spack shortcuts to
standard Python features.  I program a lot of Python, and in every case I
use "os.environ".  Using "env" instead in Spack may be more readable.  But
"os.environ" is more writable, because I don't have to look it up.  For
that reason, I am dubious of the value of these shortcuts.


On Thu, Apr 28, 2016 at 1:19 PM, Todd Gamblin <notifications@github.com>
wrote:

> In var/spack/repos/builtin/packages/nccmp/package.py:
>
> > +
> > +    version('1.8.2.0', '81e6286d4413825aec4327e61a28a580')
> > +
> > +    depends_on('netcdf')
> > +
> > +    def install(self, spec, prefix):
> > +        # Configure says: F90 and F90FLAGS are replaced by FC and
> > +        # FCFLAGS respectively in this configure, please unset
> > +        # F90/F90FLAGS and set FC/FCFLAGS instead and rerun configure
> > +        # again.
> > +        os.environ['FC'] = os.environ['F90']
> > +        del os.environ['F90']
> > +        try:
> > +            os.environ['FCFLAGS'] = os.environ['F90FLAGS']
> > +            del os.environ['F90FLAGS']
> > +        except KeyError:    # There are no flags
>
> Can you change it to this?
>
> if 'F90FLAGS' in env:
>     env['FCFLAGS'] = env['F90FLAGS']
>     del env['F90FLAGS']
>
> env is a shortcut for os.environ that you can use in Spack packages for
> readability. Also I don't really think the KeyError is more pythonic --
> the fact that it requires a comment to say why should tell you that. The
> example above is in the container implementation, so they use internal
> mechanisms to do this, but client code is intended to use the more readable a
> in b.
>
> I would argue that Ben's example is more about semantics... if you expect
> the absence of a key to be *excpetional*, then use an exception. But in
> this case it's really conditional -- the package is intended to work either
> way based on whether a key is present so it seem much more natural to me to
> write that condition out.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/LLNL/spack/pull/848/files/1a585a6c748d9445ff691e766efba350259364b5#r61466854>
>